# Repository for BIOC3001 (Advanced Practical in Molecular Biology) 

## Organisation of this Repo
The repository contains the following directories:  
* 'Scripts': QIIME scripts developed for the meta analysis of 16S rRNA sequencing data in the Cartesius supercomputer. 
* 'Data': data files used for processing and downstream analysis. 
* 'Results': text files with results obtained from running the scripts with the sequencing data.
## 1. Obtaining the data
All the files used for data analyses including the raw sequencing data and the mapping file can be downloaded from the Data folder. Remember to unzip the sequencing data with the ```gzip``` command. 
### Mapping file (tab-delimited .txt)
This file contains all the information about the samples that was used to perform the data analysis. The sample ID, barcode sequence and metadata categories are provided in different columns. The potassium, phosphorus and nitrogen measurements were transformed into an exponential scale in order to be able to perform statistical testing. 
## 2. Validate the mapping file 
We ensured that the mapping file was correctly formated with ```validate_mapping_file.py```. This script will print a message indicating whether problems were found on the file and generate a log file and a file with the location of the errors. 
## 3. Demultiplex and quality filter the reads
In this step the reads were assigned to samples based on their barcode (demultiplexing). This step also performed quality filtering based on the parameters provided. To perform these steps we used the ```split_libraries.py``` script.
## 4. Closed reference OTU picking and tree building
### Step 1. Pick OTUs
Here we run the ```pick_closed_reference_otus.py``` script, wich picks OTUs using closed reference and constructs an OTU table. OTU picking is done against a reference database, e.g. SILVA, and those sequences that do not match the database are discarded from the analysis. 
### Step 2. Summarise OTU table
To view the statistics of the OTU table we run:
```biom summarize-table -i otus/otu_table.biom```
The following results were generated from closed reference OTU picking against the SILVA 128 database:
```Num samples: 30
Num observations: 16176
Total count: 4617705
Table density (fraction of non-zero values): 0.258
Counts/sample summary:
 Min: 640.0
 Max: 1467043.0
 Median: 110746.500
 Mean: 153923.500
 Std. dev.: 248373.010
 Sample Metadata Categories: None provided
 Observation Metadata Categories: taxonomy
Counts/sample detail:
515rcbc20: 640.0
515rcbc36: 34859.0
515rcbc8: 41831.0
515rcbc13: 60071.0
515rcbc34: 62599.0
515rcbc37: 62873.0
515rcbc9: 66575.0
515rcbc27: 70604.0
515rcbc23: 74817.0
515rcbc35: 89390.0
515rcbc32: 92499.0
515rcbc11: 98214.0
515rcbc33: 105672.0
515rcbc15: 106731.0
515rcbc17: 107850.0
515rcbc18: 113643.0
515rcbc28: 114903.0
```
Alternatively --qualitative can be passed to get the number of OTUs per sample.
### Step 3. Pick a representative sequence from each OTU
Since an OTU can be comprised of many different sequences, a representative sequence of each OTU was picked for downstream analysis; this rep sequence was then used for phylogenetic alignment and taxonomy identification. The script ```pick_rep_seq.py``` was used at this step.
### Step 4. Align OTU representative sequences and filter the alignment 
Alignment was performed using PyNAST with the script: ```align_seqs.py```. Before building the tree, it is necessary to filter the alignment generated by removing columns comprised of only gaps and locations with excesive variability. Used ```filter_alignment.py``` to do this. 
### Step 5. Build a phylogenetic tree
The filtered alignment produced above was used to generate a phylogenetic tree using FASTTREE with the script ```make_phylogeny.py```. The netwick tree file was then visualised with the FigTree visualisation software. This tree is necessary for diversity measures, e.g. UniFrac. 
![tree](https://github.com/lebrusite/BIOC3301_code/blob/master/Results/phylo_tree.png)
## Taxonomy analysis 
### Assign taxonomy to representative OTU sequences
Taxonomy was assigned to each rep OTU sequence using an RDP classifier using the script: ```assign_taxonomy.py```. A text file is generated, which contains the taxonomic assingment for each OTU. For some OTUs the assignment was possible at the species level but for others this was not possible.
### Group OTUs by taxonomic composition 
OTUs were grouped by phylum with the script ```summarize_taxa.py```. This script allows to generate a new table at the phylum level, which contains the relative abundances of taxa within each sample. 
### Correlation between taxonomy and metadata categories
 The script ```observation_metadata_correlation.py``` was used to calculate the correlation between taxonomy and some of the metadata categories (pH, nitrogen, potassium, phosphorus). 
## Alpha and beta diversity analysis
The workflow script ```core_diversity_analyses.py``` was used to determine the alpha and beta diversity measures for the samples. 
### Alpha diversity correlation
The alpha diversity of sample groupings (from metadata categories) was compared with the ```compare_alpha_diversity.py``` script. In this case, a non-parametic t-test with 9999 permuations was used. 
### Compare categories
The ```compare_categories.py``` script was used to analyse the streght and significance of sample groupings using the UniFrac distance matrix (weightened or unweightened) as the input. In this case we used adonis testing, a non parametric method that partitions a distance matrix among sources of variation in order to describe the significance that a variable has in determining variation of distances. 
### Principal component analysis 2D plots 
Principal Coordinates Analysis is a technique that helps visualise a few components of variation from complex sets of data. The principal coordinates can be plotted in 2 or 3 dimensions to provide an intuitive visualisation of differences between samples. As the Emperor 3D plots generated by core_diversity_analysis.py can be difficult to visualise, the ```make_2d-plots.py``` script can be used to generate to 2d plots for better visualisation of differences between samples. 
